packages/expo-image/android/src/main/java/expo/modules/image/blurhash/BlurhashModelLoader.kt

```diff
@@ -5,13 +5,13 @@ import android.net.Uri
 import com.bumptech.glide.load.Options
 import com.bumptech.glide.load.model.ModelLoader
 import com.bumptech.glide.signature.ObjectKey
-import expo.modules.image.GlideBlurhashModel
+import expo.modules.image.BlurhashModelProvider
 
-class BlurhashModelLoader : ModelLoader<GlideBlurhashModel, Bitmap> {
-  override fun handles(model: GlideBlurhashModel): Boolean = true
+class BlurhashModelLoader : ModelLoader<BlurhashModelProvider, Bitmap> {
+  override fun handles(model: BlurhashModelProvider): Boolean = true
 
   override fun buildLoadData(
-    model: GlideBlurhashModel,
+    model: BlurhashModelProvider,
```

Comment:
seems kinda confusing – why is the provider assigned to a model? 😅

Reply:
I totally agree. The confusion arises from the fact that `BlurhashModelProvider` serves as both a model provider and a model

Initially, it acts as an abstraction between the source map and the glide model. Also, it's used as glide model - that provider returns itself, which sounds stupid. It would be best to address this in a separate pull request, as I anticipate it will require more extensive changes.